feat(f36-ai-components): add f36-ai-components package

elliotmassen opened the pull request:
## Summary

- Introduces the `@contentful/f36-ai-components` package — a collection of components for building AI conversational interfaces (chat layouts, message rendering, input with mentions, reasoning indicators, side panels, and more)
- Merges FormaV6 changes from `main` (React 19, Emotion v11 migration) into the ai-components branch
- Migrates all 28 files from `emotion` v10 to `@emotion/css` v11
- Bumps all `@contentful/f36-*` dependencies to `^6.0.0` and updates React peer deps to `^18.3.0 || >=19.1.0`
- Fixes React 19 type compatibility (`blockquote` HTMLQuoteElement typing, `Editor` type import from `@tiptap/core`, icon `variant` prop removal, controlled `Menu` requiring `onClose`)
- Upgrades `react-markdown` v7 → v9, `rehype-raw` v6 → v7, `remark-gfm` v3 → v4 for React 19 JSX namespace compatibility
- Updates Storybook 9 imports (`@storybook/addon-actions` → `storybook/actions`, `@storybook/react` → `@storybook/react-vite`)
- Renames internal `_Component` functions to `ComponentBase` suffix to satisfy `react-hooks/rules-of-hooks` linter
- Removes alpha publishing workflow and integrates package into the standard changeset release flow (added to `fixed` version group)

## New Components

- `AIChatLayout` — full-page chat layout with side panel support
- `AIChatMessage` — markdown-rendered user/assistant messages with action buttons
- `AIChatMessageList` — auto-scrolling message list with streaming support
- `AIChatInput` — TipTap-based rich text input with `@mention` support
- `AIChatHistory` — tabbed thread history with search
- `AIChatReasoning` — expandable reasoning/thinking indicator with animation
- `AIChatSidePanel` — collapsible side panel
- `AIChatSuggestionPill` — clickable suggestion chips
- `AIChatConversation` — conversation container
- `AIChatArtifactMessage` — artifact display in messages

## Test plan

> Chromatic deployment url: https://5fd1dda724cc620021ace8c5-shjeybujej.chromatic.com/?path=/docs/components-aichat-aichatartifactmessage--documentation
- [x] Build passes (`packages/core` and `packages/f36-ai-components`)
- [x] CI tests and lint pass
- [x] Storybook visual verification (animations, markdown, input, layout, history)
- [x] No changes outside `f36-ai-components` vs main (except `package-lock.json`)
- [ ] Verify changeset publishes correctly on merge

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/main/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

elliotmassen committed:
feat(f36-ai-components): add f36-ai-components package

elliotmassen committed:
fix: tidy up changeset updates

elliotmassen commented on scripts/changesets/changelog-generate.js:
```diff
@@ -10,12 +10,14 @@ const cwd = process.cwd();
   '@contentful/f36-i18n-utils',
   '@contentful/f36-website',
   '@contentful/f36-docs-utils',
+  '@contentful/f36-ai-components',
```
Is this correct? If so, I have added a note to the prerelease section in `RELEASES.md` to document it.

cf-remylenoir reviewed (dismissed):
Blocking until @contentful/team-ui-foundation review and v6 is released

mshaaban0 reviewed (approved):
Looks good to me 👍🏼

elliotmassen committed:
feat: style collapsed state

elliotmassen committed:
feat(AIChatLayout): fix up styles and story

elliotmassen committed:
fix: improve styling and layout

elliotmassen committed:
feat: complete functionality

elliotmassen committed:
fix: make styles pixel perfect

elliotmassen committed:
fix: add dependency

elliotmassen committed:
fix: use stying tokens where possible

elliotmassen committed:
fix: ensure buttons always stay on the right regardless of the prescene of a title or icon

elliotmassen committed:
feat: add tests

elliotmassen committed:
fix: code quality tweaks

elliotmassen committed:
fix: correct tab indexing when buttons are hidden

elliotmassen committed:
fix: support svg gradient in Safari

elliotmassen committed:
fix: remove comment

elliotmassen committed:
feat: refactor AIChatLayout to use display prop with three states (closed, collapsed, open)

elliotmassen committed:
refactor: rename 'type' prop to 'variant' in AIChatLayout for consistency

elliotmassen committed:
fix: include fixed positioning within the component

elliotmassen committed:
fix: centralise docs description

elliotmassen committed:
feat: add IconGradient component for gradient support in AIChatLayout

elliotmassen committed:
Merge pull request #3203 from contentful/PIC-694-forma36-create-aichatlayout-component-implementation

feat(f36-ai-components): add AIChatLayout component

elliotmassen committed:
Merge pull request #3203 from contentful/PIC-694-forma36-create-aichatlayout-component-implementation

feat(f36-ai-components): add AIChatLayout component

seanf-ctfl committed:
feat: add ai chat input component [PIC-707] (#3205)

elliotmassen committed:
feat: implement AIChatArtifactMessage

elliotmassen committed:
fix: update styles and story args for AIChatArtifactMessage component

elliotmassen committed:
fix: tidy up code

elliotmassen committed:
fix: update tests

elliotmassen committed:
Merge branch 'PIC-694-forma36-create-aichatlayout-component' into PIC-713-forma36-create-aichatartifactmessage-component

elliotmassen committed:
fix: resolve analyze_code failure

seanf-ctfl committed:
feat: add ai chat message component [PIC-706] (#3208)

elliotmassen committed:
Merge pull request #3206 from contentful/PIC-713-forma36-create-aichatartifactmessage-component

feat(f36-ai-components): add AIChatArtifactMessage component [PIC-713]

elliotmassen committed:
feat: implement AIChatReasoning component with collapsible functionality and sparkle icon animation

elliotmassen committed:
fix: tweak the tests

elliotmassen committed:
fix: remove testId prop from AIChatReasoningProps

elliotmassen committed:
Merge pull request #3211 from contentful/PIC-705-forma36-create-aichatreasoning-component

feat(f36-ai-components): add AIChatReasoning component [PIC-705]

elliotmassen committed:
feat: Implement AIChatHistory component

elliotmassen committed:
feat: styling tabs

elliotmassen committed:
feat: styling threads

elliotmassen committed:
fix: tweak stylign

elliotmassen committed:
fix: final tweaks

elliotmassen committed:
fix: add title attribute to thread content box

mshaaban0 committed:
chore: automate and publish ai package (#3214)

github-actions[bot] committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.1 [skip ci]

mshaaban0 committed:
fix: use token from vault to publish ai package (#3217)

mshaaban0 committed:
fix: use repo name as role

mshaaban0 committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.2 [skip ci]

mshaaban0 committed:
fix: remove registry url

mshaaban0 committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.3 [skip ci]

mshaaban0 committed:
fix: remove trailing slash from publish config

mshaaban0 committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.4 [skip ci]

mshaaban0 committed:
fix: use root .npmrc

mshaaban0 committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.5 [skip ci]

ContentfulCormac committed:
fix: [] Debug publish-ai-package-alpha.yml action

ContentfulCormac committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.6 [skip ci]

ContentfulCormac committed:
fix: [] Update publish-ai-package-alpha.yml

ContentfulCormac committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.7 [skip ci]

ContentfulCormac committed:
fix: [] Debug publish-ai-package-alpha.yml

ContentfulCormac committed:
fixL [] Debug publish-ai-package-alpha.yml

ContentfulCormac committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.8 [skip ci]

ContentfulCormac committed:
fix: [] Debug publish-ai-package-alpha.yml

ContentfulCormac committed:
fix: [] Debug publish-ai-package-alpha.yml

ContentfulCormac committed:
fix: [] Debug publish-ai-package-alpha.yml

ContentfulCormac committed:
fix: [] Debug publish-ai-package-alpha.yml

ContentfulCormac committed:
fix: [] Debug publish-ai-package-alpha.yml

ContentfulCormac committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.9 [skip ci]

ContentfulCormac committed:
fix [] Debug publish-ai-package-alpha.yml

ContentfulCormac committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.10 [skip ci]

ContentfulCormac committed:
fix: [] Debug publish-ai-package-alpha.yml

ContentfulCormac committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.11 [skip ci]

ContentfulCormac committed:
chore: [] bump version

ContentfulCormac committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.13 [skip ci]

elliotmassen committed:
fix: remove unused formatTime export from AIChatHistory

seanf-ctfl committed:
feat: add mention support to ai chat input [PIC-710] (#3212)

seanf-ctfl committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.14 [skip ci]

elliotmassen committed:
fix: update styles for AIChatHistory components to account for new designs

elliotmassen committed:
fix: simplify thread click handling in AIChatHistory component

elliotmassen committed:
fix: update AIChatHistory components to use NavList

elliotmassen committed:
fix: update AIChatHistoryTabs styles for active tab indicators

elliotmassen committed:
fix: add tooltips for status icons in AIChatHistory

elliotmassen committed:
Merge pull request #3216 from contentful/PIC-709-forma36-create-aichathistory-component

feat(f36-ai-components): add AIChatHistory component

elliotmassen committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.15 [skip ci]

elliotmassen committed:
feat: outline of header animation

elliotmassen committed:
fix: some tweaks most of the way there functionally

elliotmassen committed:
fix: clip header slider to remove button overlap

elliotmassen committed:
fix: use icon gradient in story

elliotmassen committed:
feat: something that works

elliotmassen committed:
feat: integrate Slider component for AIChatLayout story

elliotmassen committed:
feat: add support for button positioning in AIChatLayout with buttonsStart and buttonsEnd

elliotmassen committed:
fix: final styling tweaks

elliotmassen committed:
feat: refactor AIChatLayout to use header prop for title, icon, and buttons

elliotmassen committed:
feat: enhance AIChatLayout and Slider components with improved button handling and state comparison

elliotmassen committed:
fix: tidy up button usage

elliotmassen committed:
feat: update mock chat messages with placeholder content for testing

elliotmassen committed:
feat: rename Basic story to Default and add WithChangableHeader story with dynamic props

elliotmassen committed:
fix: make fixed buttons zero-to-many

elliotmassen committed:
fix: final tweaks

elliotmassen committed:
fix: tidy up slider

elliotmassen committed:
fix: resolve import issue with storybook actions in mock data

elliotmassen committed:
fix: simplify header identification

elliotmassen committed:
fix: update data-testid attribute syntax and add comprehensive tests for Slider component

elliotmassen committed:
fix: refactor Slider component to use slideKey and update tests accordingly

elliotmassen committed:
fix: remove tests for missing slideKey and children in Slider component

elliotmassen committed:
fix: adjust padding and margin values in AIChatHistory styles for consistency

elliotmassen committed:
feat: add AIChatSidePanel component

elliotmassen committed:
Merge pull request #3224 from contentful/PIC-734-forma36-update-ai-chat-layout-toolbar-to-support

feat(f36-ai-components): Update AIChatLayout to support multiple headers

elliotmassen committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.16 [skip ci]

seanf-ctfl committed:
fix: fix null pointer error on list item when is children is not defined [PIC-706] (#3227)

seanf-ctfl committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.17 [skip ci]

seanf-ctfl committed:
fix: fix null pointer error on list item when is children is not defined [PIC-706] (#3229)

seanf-ctfl committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.18 [skip ci]

elliotmassen committed:
fix(f36-ai-components): re-render AIChatReasoning when children update (#3233)

* fix: update AIChatReasoning to recalculate height on children change and add dynamic children story

* fix: resolve flickering icons after collapsing

* fix: remove unneccessary comment

mshaaban0 committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.19 [skip ci]

nikunj-tilva committed:
feat: exporting primary gradient icon [PIC-747]

nikunj-tilva committed:
chore: package-lock update [PIC-747]

nikunj-tilva committed:
chore: syncing build node version and nvmrc node version [PIC-747]

nikunj-tilva committed:
chore: syncing build node version and nvmrc node version for other steps [PIC-747]

nikunj-tilva committed:
Merge pull request #3240 from contentful/PIC-747-export-ai-icon-gradient

feat: exporting AI Gradient icons

nikunj-tilva committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.20 [skip ci]

nikunj-tilva committed:
feat: exposing svg gradient styles [PIC-747]

nikunj-tilva committed:
Merge pull request #3242 from contentful/PIC-747-ai-icon-gradient-styles

feat: exposing svg gradient styles [PIC-747]

nikunj-tilva committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.21 [skip ci]

mshaaban0 committed:
feat: add AIChatEmptyState component to ai-components package [PIC-746] (#3243)

* feat: add AIChatEmptyState component

* feat: group AIChat components stories

* feat: add AIChatEmptyState story

* feat: refactored AIChatEmptyState to AIChatConversationEmptyState [PIC-746]

* feat: lint-fix [PIC-746]

* refactor: AIConversationEmptyState types export

---------

Co-authored-by: Nikunj Tilva <nikunj.tilva@contentful.com>

mshaaban0 committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.22 [skip ci]

Rowadz committed:
fix: chat bubble alignments

Rowadz committed:
Merge branch 'PIC-694-forma36-create-aichatlayout-component' into fix/chat-bubble-alignments

Rowadz committed:
Merge pull request #3244 from contentful/fix/chat-bubble-alignments

fix: chat bubble alignments

Rowadz committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.23 [skip ci]

elliotmassen committed:
fix(f36-ai-components): combine AIChatArtifactMessage className with internal style

elliotmassen committed:
Merge pull request #3245 from contentful/PIC-721-user-interface-display-aichat-artifacts-in-aicha

fix(f36-ai-components): combine AIChatArtifactMessage className with internal style

elliotmassen committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.24 [skip ci]

nikunj-tilva committed:
feat: added ChatHistoryEmptyState component [PIC-783]

nikunj-tilva committed:
feat: updated styles for ChatHistoryEmptyState component [PIC-783]

elliotmassen committed:
feat(f36-ai-components): add AIChatMessageList component

nikunj-tilva committed:
feat: extending common props [PIC-783]

jamesmbourne committed:
chore: [] ignore npm scripts

jamesmbourne committed:
chore: [] ignore npm scripts

elliotmassen committed:
feat(f36-ai-components): enhance AIChatMessageList with scroll behavior

elliotmassen committed:
fix(f36-ai-components): add main panel component and tweak styles for message list integration

elliotmassen committed:
fix(f36-ai-components): rename AIChatMainPanel to AIChatConversation

elliotmassen committed:
fix(f36-ai-components): tweak message styles

nikunj-tilva committed:
Merge pull request #3250 from contentful/PIC-783-chat-history-emptystate-component

feat: added chat history empty state component [PIC-783]

elliotmassen committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.25 [skip ci]

cf-remylenoir committed:
chore(cdn): add `pre-build` script in `build`

elliotmassen committed:
feat(f36-ai-components): add WithStreamingChunks story for AIChatMessageList with streaming behavior

elliotmassen committed:
Merge branch 'PIC-694-forma36-create-aichatlayout-component' into PIC-762-forma36-create-aichatmessagelist-component

elliotmassen committed:
Merge pull request #3254 from contentful/PIC-762-forma36-create-aichatmessagelist-component

feat(f36-ai-components): add AIChatMessageList component

elliotmassen committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.26 [skip ci]

elliotmassen committed:
Merge branch 'main' into PIC-694-forma36-create-aichatlayout-component

elliotmassen committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.27 [skip ci]

elliotmassen committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.28 [skip ci]

nikunj-tilva committed:
feat: added missing markdown tags AIChatMessage [PIC-764]

nikunj-tilva committed:
feat: fixed codeblock markdown issue AIChatMessage [PIC-764]

nikunj-tilva committed:
feat: spacing adjustments [PIC-764]

nikunj-tilva committed:
feat: spacing adjustments [PIC-764]

nikunj-tilva committed:
Merge branch 'PIC-694-forma36-create-aichatlayout-component' into PIC-AI-chat-message-markdown

nikunj-tilva committed:
feat: spacing adjustments [PIC-764]

nikunj-tilva committed:
Merge pull request #3265 from contentful/PIC-AI-chat-message-markdown

feat: ai chat message markdown improvements [PIC-764]

nikunj-tilva committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.29 [skip ci]

nikunj-tilva committed:
feat: paragraph marging adjustments [PIC-764]

nikunj-tilva committed:
Merge pull request #3267 from contentful/PIC-764-remove-margin-last-paragraph

feat: paragraph marging adjustments [PIC-764]

nikunj-tilva committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.30 [skip ci]

nikunj-tilva committed:
feat: added AIChatMentionList auto placement support [PIC-758]

nikunj-tilva committed:
feat: handling scrolling behavior for AIChatMentionList position [PIC-758]

nikunj-tilva committed:
Merge pull request #3269 from contentful/PIC-758-ai-chat-mention-positioning

feat: added AIChatMentionList auto placement support [PIC-758]

nikunj-tilva committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.31 [skip ci]

elliotmassen committed:
chore: document other use of layout variant

elliotmassen committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.32 [skip ci]

elliotmassen committed:
feat: implement AIChatSuggestionPill component with styles and tests

elliotmassen committed:
feat: enhance AIChatSuggestionPill styles

elliotmassen committed:
feat: add transition effect to AIChatSuggestionPill icon and update styles

elliotmassen committed:
feat: extend AIChatSuggestionPillProps with CommonProps and clean up prop definitions

toniwowtscherk committed:
Update AIChatSuggestionPill.styles.ts

Change top and bottom padding to 6px instead of 5px to match the design

elliotmassen committed:
Merge pull request #3275 from contentful/PIC-833

feat(f36-ai-components): add AIChatSuggestionPill component [PIC-833]

elliotmassen committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.33 [skip ci]

seanf-ctfl committed:
fix: Re-Add bugfix for AiChatMessage markdown li [PIC-845] (#3280)

seanf-ctfl committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.34 [skip ci]

seanf-ctfl committed:
feat: AIChatMessage add ability for clients to add advanced components in messages [PIC-847] (#3282)

seanf-ctfl committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.35 [skip ci]

elliotmassen committed:
feat: add height and position props to AIChatLayout for flexible layout control

elliotmassen committed:
fix(AIChat): improve relative auto height implementation and story

elliotmassen committed:
Merge pull request #3297 from contentful/PIC-879

feat(AIChat): add height and position props to AIChatLayout for flexible layout control

elliotmassen committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.36 [skip ci]

seanf-ctfl committed:
feat: Add AI Runs List Loading View [PIC-805] (#3326)

seanf-ctfl committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.37 [skip ci]

nikunj-tilva committed:
Merge remote-tracking branch 'origin/main' into PIC-694-forma36-create-aichatlayout-component

nikunj-tilva committed:
chore: update dependencies to f36 v6 and refactor emotion imports to @emotion/css

nikunj-tilva committed:
chore: update dependencies and improve AI chat components

- Added @tiptap/core dependency to package.json.
- Updated AIChatHistoryEmptyState to use tokens for icon color.
- Refactored AIChatInput to import Editor type correctly.
- Modified AIChatMessage to use any type for blockquote props.
- Changed import paths for action from
 '@storybook/addon-actions' to 'storybook/actions' in multiple story files.

nikunj-tilva committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.38 [skip ci]

nikunj-tilva committed:
chore: syncing changes with main for outside f36 components changes [PIC-694]

nikunj-tilva committed:
chore: bump @contentful/f36-ai-components to 0.0.1-alpha.39 [skip ci]

nikunj-tilva committed:
chore: removed alpha publishing workflow for ai-components [PIC-694]

nikunj-tilva committed:
chore: package-lock update [PIC-694]

nikunj-tilva committed:
chore: package-lock update [PIC-694]

nikunj-tilva committed:
chore: package-lock update [PIC-694]

nikunj-tilva committed:
style: fix indentation in Slider.styles.ts for better readability [PIC-694]

nikunj-tilva committed:
refactor: rename internal components to use 'Base' suffix for linter [PIC-949]

- using @storybook/react-vite instead of @storybook/react

nikunj-tilva committed:
chore: update dependencies and fix AIChatMentionList menu behavior

- Updated react-markdown to version 9.0.3, rehype-raw to version 7.0.0,
and remark-gfm to version 4.0.0 in package.json.
- Modified AIChatMentionList component to include
onClose handler for the Menu component.
- Changed Storybook import type from '@storybook/react/types-6-0' to '@storybook/react-vite'
in AIChatArtifactMessage stories.

mshaaban0 reviewed (approved):
LGTM

Lelith commented on packages/f36-ai-components/src/AIChatHistory/AIChatHistoryTabs/AIChatHistoryTabs.test.tsx:
```diff
@@ -0,0 +1,87 @@
+import { MessageGroups } from '../AIChatHistory';
+import { AIChatHistoryTabs } from './AIChatHistoryTabs';
+
+describe('AIChatHistoryTabs', () => {
```
:accessibility: misses an axe test

Lelith reviewed (changes requested):
I did not have much time for reviewing, but I see some common themes:

**Blocking issues:** 
* We need to enable all components for internationalization - therefore no hardcoded strings are allowed
* Tests for components have to include an axe acessibility test

**General Improvement**
* We are using the `Box`component quite a lot where other components like `Flex`or `Stack` should have been preferred

denkristoffer committed:
chore: tiptap v3 [] (#3344)

* chore: tiptap v3

nikunj-tilva committed:
Merge branch 'main' into PIC-694-forma36-create-aichatlayout-component

nikunj-tilva committed:
Merge branch 'main' into PIC-694-forma36-create-aichatlayout-component

nikunj-tilva commented:
Merging this considering this ticket for followup https://contentful.atlassian.net/browse/PIC-1013
